### PR TITLE
logging: add backend for xtensa simulator

### LINKED ISSUE #10164: logger sample fails on qemu_xtensa due to lack of backend
Getting no output on both, see PR #10163 and the CI results.

### PR DESCRIPTION
**Pull request by nashif**
Add backend for the xtensa simulator.

Fixes #10164

### CODE REVIEW

File: subsys/logging/log_backend_xtensa_sim.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+				: "a" (a2), "a" (a3), "a" (a4), "a" (a5)
+				: "memory");
+}
+
+static u8_t buf[_STDOUT_BUF_SIZE];
+
+static int char_out(u8_t *data, size_t length, void *ctx)
+{
+	for (size_t i = 0; i < length; i++) {
+		preprint_char(data[i]);
```
Comment (jcmvbkbc) [inline subsys/logging/log_backend_xtensa_sim.c]: write simcall can output the whole buffer in one invocation

Comment (nashif) [reply subsys/logging/log_backend_xtensa_sim.c]: thanks, addressed.

File: subsys/logging/log_backend_xtensa_sim.c [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+#include <stdio.h>
+#include <stddef.h>
+#include <logging/log_backend.h>
+#include <logging/log_core.h>
+#include <logging/log_msg.h>
+#include <logging/log_output.h>
+#include <xtensa/simcall.h>
+
+#define _STDOUT_BUF_SIZE 256
```
Comment (nordic-krch) [inline subsys/logging/log_backend_xtensa_sim.c]: if there a potential RAM shortages on xtensa boards this should be moved to kconfig.

**Review by nordic-krch — APPROVED** [review]
moving buffer size to kconfig would be nice.

**Pull request merged**